docs/spec/src/integration/spec/3-data-structs.md

```diff
@@ -126,7 +126,60 @@ Users who want to pay-per-blob need to set the cumulative_payment. `timestamp` i
+
+### BlobKey (Blob Header Hash)
+
+The `blobKey` (also known as `blob_header_hash` or `blobHeaderHash`) is a unique identifier for a blob dispersal in the EigenDA system. It is computed as the keccak256 hash of the ABI-encoded `BlobHeader`, and is cryptographically equivalent to the `blob_header_hash` used in on-chain verification. This identifier serves as the primary lookup key throughout the EigenDA system.
+
+#### Computing the BlobKey
+
+The computation differs between V1 and V2 blob headers. For V2 BlobHeaders, the hashing follows a nested structure:
+
+```solidity
+blobKey = keccak256(
+    abi.encode(
+        keccak256(abi.encode(blobHeader.version, blobHeader.quorumNumbers, blobHeader.commitment)),
+        blobHeader.paymentHeaderHash
+    )
+)
+```
+
+For V1 BlobHeaders, the computation is simpler:
+
+```solidity
+blobKey = keccak256(abi.encode(blobHeader))
+```
+
+The V2 approach uses a two-level hashing structure where the core dispersal fields (version, quorum numbers, and commitment) are hashed first, then combined with the payment metadata hash. This design reflects the separation between dispersal parameters and payment concerns.
+
+When a rollup receives an encoded DA commitment from the proxy, the `blobKey` can be extracted by deserializing the BlobCertificate from the commitment payload, extracting its BlobHeader, and computing the hash as shown above.
+
+#### Relationship to Other Data Structures
+
+The diagram below illustrates how `blobKey` relates to other core EigenDA data structures:
+
+![Data Structure Relationships](../../assets/integration/blob-key-relationships.png)
+
+The `BlobHeader` is hashed to produce the `blobKey`. A `BlobCertificate` wraps a `BlobHeader` along with signature and relay keys. The `BlobInclusionInfo` contains a `BlobCertificate` and is used to prove inclusion of that certificate in a batch via a Merkle proof. The `BatchHeader` contains a `batchRoot` which is the root of the Merkle tree whose leaves are hashes of `BlobCertificate`s.
+
+#### Code References
+
+The Solidity implementations can be found in [`hashBlobHeaderV2()`](https://github.com/Layr-Labs/eigenda/blob/master/contracts/src/integrations/cert/libraries/EigenDACertVerificationLib.sol#L324) for V2 and [`hashBlobHeader()`](https://github.com/Layr-Labs/eigenda/blob/master/contracts/src/integrations/cert/legacy/v1/EigenDACertVerificationV1Lib.sol#L241) for V1.
+
+The Go implementations include [`ComputeBlobKey()`](https://github.com/Layr-Labs/eigenda/blob/master/core/v2/serialization.go#L42) for V2 computation, [`hashBlobHeader()`](https://github.com/Layr-Labs/eigenda/blob/master/api/hashing/node_hashing.go#L62) for node hashing, and [`HashBlobHeader()`](https://github.com/Layr-Labs/eigenda/blob/master/api/proxy/store/generated_key/eigenda/verify/hasher.go#L109) for proxy verification.
```

Comment:
why does the node hashing reference matter? iiuc this isn't actually computing the blob key and is instead hashing fields of the DA Cert for the disperser to sign over:
https://github.com/Layr-Labs/eigenda/blob/9f2cf022657c4cce9604bd4caf85472c4b23906a/api/clients/v2/dispersal_request_signer.go#L103-L115